_query-dsl/specialized/script-score.md

```diff
@@ -0,0 +1,332 @@
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.14384104,
+    "hits": [
+      {
+        "_index": "testindex1",
+        "_id": "1",
+        "_score": 0.14384104,
+        "_source": {
+          "name": "John Doe",
+          "multiplier": 0.5
+        }
+      }
+    ]
+  }
+}
+```
+
+## Parameters
+
+The `script_score` query supports the following top-level parameters.
+
+Parameter | Data type | Description
+:--- | :--- | :---
+`query` | Object | The query to use for search. Required.
+`script` | Object | The script to use to calculate the score of the documents that the `query` returns. Required.
+`min_score` | Float | Excludes documents with a score lower than `min_score` from the results. Optional.
+`boost` | Float | Boosts the documents' scores by the given multiplier. Values less than 1.0 decrease relevance, and values greater than 1.0 increase relevance. Default is 1.0. 
+
+The relevance scores calculated by the `script_score` query cannot be negative. 
+{: .important}
+
+## Customizing score calculation with built-in functions
+
+To customize score calculation, you can use one of the built-in Painless functions. For every function, OpenSearch provides one or more Painless methods you can access in the script score context. You can call the Painless methods listed in the following sections directly, without using a class name or instance name qualifier.
+
+### Saturation
+
+The saturation function calculates saturation as `score = value /(value + pivot)`, where `value` is the field value and `pivot` is chosen so that the score is greater than 0.5 if `value` is greater than `pivot` and less than 0.5 if `value` is less than `pivot`. The score is in the (0, 1) range. To apply a saturation function, call the following Painless method:
+
+- `double saturation(double <field-value>, double <pivot>)`
+    
+#### Example
+
+The following example searches for the text `neural search` in the `articles` index. It combines the original document relevance score with the `article_rank` value, which is transformed with a saturation function first:
+
+```json
+GET articles/_search
+{
+  "query": {
+    "script_score": {
+      "query": {
+        "match": { "article_name": "neural search" }
+      },
+      "script" : {
+        "source" : "_score + saturation(doc['article_rank'].value, 11)"
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+### Sigmoid
+
+Similarly to the saturation function, the sigmoid function calculates the score as `score = value^exp/ (value^exp + pivot^exp)`, where `value` is the field value, `exp` is an exponent scaling factor, and `pivot` is chosen so that the score is greater than 0.5 if `value` is greater than `pivot` and less than 0.5 if `value` is less than `pivot`. To apply a sigmoid function, call the following Painless method:
```

Comment:
```suggestion
Similarly to the saturation function, the sigmoid function calculates the score as `score = value^exp/ (value^exp + pivot^exp)`, where `value` is the field value, `exp` is an exponent scaling factor, and `pivot` is chosen so that the score is higher than 0.5 if `value` is higher than `pivot` and lower than 0.5 if `value` is lower than `pivot`. To apply a sigmoid function, call the following Painless method:
```